OSDOCS12386: Add graceful reboot module to WMCO docs (#80780)
openshift/openshift-docs: modules/nodes-nodes-rebooting-gracefully.adoc

//
// * nodes/nodes-nodes-rebooting.adoc

ifeval::["{context}" == "nodes-nodes-rebooting"]
:nodes:
endif::[]
ifeval::["{context}" == "enabling-windows-container-workloads"]
:windows:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="nodes-nodes-rebooting-gracefully_{context}"]
= Rebooting a node gracefully

ifdef::windows[]

> The Windows Machine Config Operator (WMCO) minimizes node reboots whenever possible. However, certain operations and updates do require a reboot to ensure that changes are applied correctly and securely. To safely reboot your Windows nodes, you can use the graceful reboot process as needed. For inforamation on gracefully rebooting a standard {product-title} node, see "Rebooting a node gracefully" in the Nodes documentation. 

Reviewer: ```suggestion
The Windows Machine Config Operator (WMCO) minimizes node reboots whenever possible. However, certain operations and updates require a reboot to ensure that changes are applied correctly and securely. To safely reboot your Windows nodes, use the graceful reboot process. For information on gracefully rebooting a standard {product-title} node, see "Rebooting a node gracefully" in the Nodes documentation. 
```

$ sudo systemctl reboot
----
====
endif::nodes[]
ifdef::windows[]

> . SSH into the Windows node and enter PowerShell by using the following command:

Reviewer: ```suggestion
. SSH into the Windows node and enter PowerShell by running the following command:
```
Just for consistency with your other steps.

+
[source,terminal]
----
C:\> powershell
----

. Restart the node by running the following command:
+
[source,terminal]
----
C:\>  Restart-Computer -Force
----

. Windows nodes on Amazon Web Services (AWS) do not return to `READY` state after a graceful reboot due to an inconsistency with the EC2 instance metadata routes and the Host Network Service (HNS) networks.
+
After the reboot, SSH into any Windows node on AWS and add the route by running the following command in a shell prompt:
+
[source,terminal]
----
C:\> route add 169.254.169.254 mask 255.255.255.0 <gateway_ip>
----
+

> where:

Reviewer: ```suggestion
In the above command:
```
Maybe add something of that sort rather than just the word "where".

Author: @jneczypor This is [standard formatting for additional information on what a line of a code block represents](additional information on what a line of a code block represents).

+
--
`169.254.169.254`:: Specifies the address mask of the EC2 instance metadata endpoint.

> `255.255.255.255`:: Specifies the network mask of the EC2 instance metadata endpoint.

Reviewer 2: `255.255.255.255`, I'm not sure it's a good example, I prefer `255.255.255.0`

> 255.255.255.255 is a valid address mask, but it serves a special purpose.
> 
> 255.255.255.255 is a unique subnet mask that specifies a single IP address. This means all bits are set to network bits, with no host bits available. In this context, the IP address corresponds to a broadcast address, also known as a "limited broadcast." When a packet uses this mask for broadcasting, it sends data to all hosts on the same network segment.
> Typically, this mask is not used for actual network segmentation but rather for specific scenarios, such as transmitting broadcast packets.

> `<gateway_ip>`:: Specifies the corresponding IP address of the gateway in Windows instance, which you can find by running the following command:

Reviewer: ```suggestion
`<gateway_ip>`:: Specifies the corresponding IP address of the gateway in the Windows instance, which you can find by running the following command:
```
